doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -766,58 +726,81 @@ the conversion in hypotheses :n:`{+ @ident}`.
          pattern (0 = 0).
          fold not.
 
-   .. tacv:: fold {+ @term}
+.. tacn:: pattern {+, @pattern_occs } {? @occurrences }
 
-      Equivalent to :n:`fold @term ; ... ; fold @term`.
+   .. todo PR: how do multiple pattern_occs work with @occurrences??
 
-.. tacn:: pattern @term
-   :name: pattern
+   Performs :term:`eta-expansion`.  If :n:`@occurrences` is given,
+   the specified occurrences will be replaced in the selected hypotheses and/or goal.
+   If :n:`@occurrences` is not given, every occurrence of the resulting term in the goal is
+   of the current goal (say :g:`T`)
+   for a set of :n:`@one_term`\s in the :n:`@pattern_occs` by:
 
-   This command applies to any goal. The argument :n:`@term` must be a free
-   subterm of the current goal. The command pattern performs :math:`\beta`-expansion
-   (the inverse of :math:`\beta`-reduction) of the current goal (say :g:`T`) by
+   + replacing all selected occurrences of the :n:`@one_term`\s in :g:`T` with fresh variables
+   + abstracting these variables
+   + applying the abstracted goal to the :n:`@one_term`\s
 
-   + replacing all occurrences of :n:`@term` in :g:`T` with a fresh variable
-   + abstracting this variable
-   + applying the abstracted goal to :n:`@term`
+   The :n:`@one_term`\s must be free subterms of the current goal.
+
+   Starting from a goal :n:`G(t__1, …, t__n)`, :tacn:`pattern`
+   :n:`t__1, …, t__n` generates the equivalent goal
+   :n:`(fun (x__1:A__1 … (X__n:A__n) => G(x__1 … x__n)) t__1 … t__n`.
+   If :g:`t`:sub:`i` occurs in one of the generated types :g:`A`:sub:`j` these
+   occurrences will also be considered and possibly abstracted.
```

Comment:
It may be worth noting that this is only for `j > i` (if `i <= j`, the resulting function would not be a valid term because you cannot mention later function arguments in the types of earlier ones)